cluster-sync/sync.sh

```diff
@@ -123,6 +123,25 @@ function wait_cdi_pods_updated {
   fi
 }
 
+# Setup some datavolumes in older version for testing upgrades
+# Done unconditionally to make it easier to write tests.
+function setup_for_upgrade_testing {
+  if _kubectl get namespace cdi-testing-old-version-artifacts ; then
+    echo "Old version testing environment already setup"
+    return
+  fi
+  echo "Missing old version environment setup, creating"
```

Comment:
why is that missing? that should be done in preparation of test infrastructure

Reply:
It is missing in the first time this function is called, I added the messages for debug/legibility purposes.

Reply:
I still do not get why it can be missing, it is all created in lines 149-152 in steps for setup: `test-infra`

Reply:
I agree regarding the first 3 applies but here we also apply the new `upgrade-testing-artifacts.yaml` and wait for Readiness.

Reply:
Yes, it's about upgrade-testing-artifacts.yaml and setting up a quick environment so that I can setup some easy DVs (I didn't want to rely on external URLs)

Reply:
so if I read the script correctly, previously we have only applied the 3 manifests (cdi-testing-sa, file-host, registry-host) after checking this condition 
```
if [ "${KUBEVIRT_PROVIDER}" != "external" ] && [ "${CDI_SYNC}" == "test-infra" ]; then
```
and now we are going to apply them also for `external` provider if we do `upgrade` tests. Is that ok?

Reply:
I can introduce the external provider limitations